src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -107,6 +111,14 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
+    await this.startConnectingInternal(reconnecting);
+    return await new Promise<void>((resolve, reject) => {
+      this.bootstrapResolved = resolve;
```

Comment:
In a slow (or almost offline) internet connection, the `startConnecting` will take a while to resolve. I wonder if we need to add a timeout or retry logic.

Also, if customers rely on the existing behavior, they may experience unexpected user experiences. For example, if customers update the UI after `await messagingSession.start()`, they will see some delay now.

Reply:
This seems like an existing issue even without this change. I agree that a timeout or retry logic shall be helpful but I think that could be tackled separately considering this fix is a result of a customer ticket.

Reply:
Right. The `startConnecting` previously called other async APIs like `getMessagingSessionEndpoint`. `getMessagingSessionEndpoint` will be slow too.